Tasks/Common/VstsAzureRestHelpers_/VstsAzureRestHelpers_.psm1

```diff
@@ -153,7 +162,7 @@ function Get-SpnAccessToken {
     try
     {
         $proxyUri = Get-ProxyUri $authUri
-        if (($proxyUri -eq $null) -or ($proxyUri.AbsoluteUri -eq $null) -or ($proxyUri.AbsoluteUri -eq $authUri))
+        if (($proxyUri -eq $null) -or ($proxyUri.AbsoluteUri -eq $null) -or ($authUri -like "*$($proxyUri.Host)*"))
```

Comment:
the wildcard comparison looks susceptible to false positives. is an accurate determination not possible?

also since you do this similar host comparison in multiple places, would it make more sense for `get-proxyuri` to perform the comparison and return null.